core/payment-driver/zksync/src/zksync/wallet.rs

```diff
@@ -315,35 +345,29 @@ pub async fn withdraw<S: EthereumSigner + Clone, P: Provider + Clone>(
     recipient: Option<String>,
 ) -> Result<SyncTransactionHandle<P>, GenericError> {
     let token = get_network_token(network, None);
-    let balance = wallet
-        .get_balance(BlockStatus::Committed, token.as_str())
-        .await
-        .map_err(GenericError::new)?;
+    let balance = get_balance(&wallet, &token).await?;
     info!(
         "Wallet funded with {} {} available for withdrawal",
         utils::big_uint_to_big_dec(balance.clone()),
         token
     );
 
     info!("Obtaining withdrawal fee");
-    let address = wallet.address();
-    let withdraw_fee = wallet
-        .provider
-        .get_tx_fee(TxFeeTypes::Withdraw, address, token.as_str())
-        .await
-        .map_err(GenericError::new)?
-        .total_fee;
+    let withdraw_fee = get_withdraw_fee(&wallet, &token).await?;
     info!(
         "Withdrawal transaction fee {:.5} {}",
         utils::big_uint_to_big_dec(withdraw_fee.clone()),
         token
     );
+    if withdraw_fee > balance {
+        return Err(GenericError::new("Not enough balance to withdraw"));
```

Comment:
maybe print balance + required amounts in this error too?

Reply:
This error should actually never be raised because it's already checked in `exit`. This is just to double-check.

Reply:
From exit you will get nice messages like this:
```
Error: Not enough balance to exit. Minimum required balance to pay withdraw fees is 0.02957 tGLM
```

Reply:
Neat 👍